appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -161,6 +172,43 @@ public void onPropertyChange(String propertyName, String newValue) {
         }
     }
 
+    public void addCSVFile(MockNonVisibleComponent csvSource) {
+        if (!csvSource.getType().equals("CSVFile")) {
+            return;
+        }
+
+        if (csvSource.properties.hasProperty("SourceFile")) {
+            String fileSource = csvSource.properties.getPropertyValue("SourceFile");
+
+
+            long projectId = Ode.getInstance().getCurrentYoungAndroidProjectId();
+
+            Ode.getInstance().getProjectService().load(projectId, "assets/" + fileSource, new AsyncCallback<String>() {
+                @Override
+                public void onFailure(Throwable caught) {
+                    ErrorReporter.reportError(caught.getMessage());
+                }
+
+                @Override
+                public void onSuccess(String result) {
+                    MockChart.this.onSelectedChange(true); // otherwise the last imported component
+
+                    String[] columnNames = result.split("\n")[0].split(",");
```

Comment:
Use CSV parsing utilities

Reply:
Move CSVParser to shared package

Reply:
Cannot move to shared package due to I/O operations not being usable on the client.

Reply:
Solution: add callable parseCSV in ProjectService. Also, possibly refactor the CSVFile component to contain it's contents in the editor?

Reply:
To be handled: #95